results-loader/src/utils/results-utils.ts

```diff
@@ -0,0 +1,39 @@
+import ClientOAuth2 from "client-oauth2";
+
+const PORTAL_AUTH_PATH = "/auth/oauth_authorize";
+
+const getURLParam = (name: string) => {
+  const url = (self || window).location.href;
+  name = name.replace(/[[]]/g, "\\$&");
+  const regex = new RegExp(`[#?&]${name}(=([^&#]*)|&|#|$)`);
+  const results = regex.exec(url);
+  if (!results) return null;
+  return decodeURIComponent(results[2].replace(/\+/g, " "));
+};
+
+export const authorizeInPortal = (portalUrl: string, oauthClientName: string) => {
+  const portalAuth = new ClientOAuth2({
+    clientId: oauthClientName,
+    redirectUri: window.location.origin + window.location.pathname + window.location.search,
+    authorizationUri: `${portalUrl}${PORTAL_AUTH_PATH}`
+  });
+  // Redirect
+  window.location.href = portalAuth.token.getUri();
+};
+
+export const readPortalAccessToken = (portalUrl: string, oauthClientName: string): string => {
+  // No error handling to keep the code minimal.
```

Comment:
Maybe add a `TODO` comment here? 
I'm not sure what errors can happen at this point. The authorizeInPortal call is just constructing a URL to redirect to, so I don't think it can error. 
What is possible is that some error condition happens in the portal and it redirects back to the client with an error message.
This error redirect can happen here:
https://github.com/concord-consortium/rigse/blob/917d5aaa788502b00fe906749af80275203b83fd/rails/app/controllers/auth_controller.rb#L45-L47
The invalid response can happen for these 2 reasons:
https://github.com/concord-consortium/rigse/blob/917d5aaa788502b00fe906749af80275203b83fd/rails/app/models/access_grant.rb#L65-L74

Basically if the client is not configured correctly in the portal. Or the request from the client uses an invalid response type. The response type is handled by the ClientOAuth2 library so hopefully that would never be a problem. But if the client is not configured with the 'public' type then the portal should redirect back to the client with a `error` url param.

Reply:
For now I added these notes in a TODO comment.

Reply:
From looking at the above code that you linked, it appears that we will get an `error` URL param with a message that we can output to the user.  If I tweaked the return type of `readPortalAccessToken`, it could return an object that specifies either an error or a token.  Then we can output the error if we find one.

Reply:
I added one more commit that attempts to pull the portal error message from the URL and display that to the user.  So we have a first pass of handling errors from the portal that are communicated in the URL.